pkgs/development/compilers/vala/default.nix

```diff
@@ -91,6 +91,12 @@ let
 
     doCheck = false; # fails, requires dbus daemon
 
+    passthru = {
+      updateScript = gnome3.updateScript {
```

Comment:
Update script unfortunately does not work for Vala, due to the specific way the derivation is structured. We would need to update it like this:

```patch
--- a/pkgs/development/compilers/vala/default.nix
+++ b/pkgs/development/compilers/vala/default.nix
@@ -1,17 +1,15 @@
 { stdenv, lib, fetchurl, fetchpatch, pkgconfig, flex, bison, libxslt, autoconf, automake, autoreconfHook
-, graphviz, glib, libiconv, libintl, libtool, expat, substituteAll
+, graphviz, glib, libiconv, libintl, libtool, expat, substituteAll, gnome3
 }:
 
 let
   generic = lib.makeOverridable ({
-    major, minor, sha256,
+    version, sha256,
     extraNativeBuildInputs ? [],
     extraBuildInputs ? [],
     withGraphviz ? false
   }:
   let
-    atLeast = lib.versionAtLeast "${major}.${minor}";
-
     # Patches from the openembedded-core project to build vala without graphviz
     # support. We need to apply an additional patch to allow building when the
     # header file isn't available at all, but that patch (./gvc-compat.patch)
@@ -45,21 +43,21 @@
         #     0.42.4: https://github.com/openembedded/openembedded-core/raw/f2b4f9ec6f44dced7f88df849cca68961419eeb8/meta/recipes-devtools/vala/vala/disable-graphviz.patch
         "0.44" = ./disable-graphviz-0.44.1.patch;
 
-      }.${major} or (throw "no graphviz patch for this version of vala");
+      }.${lib.versions.majorMinor version} or (throw "no graphviz patch for this version of vala");
 
-    disableGraphviz = atLeast "0.38" && !withGraphviz;
+    disableGraphviz = lib.versionAtLeast version "0.38" && !withGraphviz;
 
   in stdenv.mkDerivation rec {
-    name = "vala-${version}";
-    version = "${major}.${minor}";
+    pname = "vala";
+    inherit version;
 
     setupHook = substituteAll {
       src = ./setup-hook.sh;
-      apiVersion = major;
+      apiVersion = lib.versions.majorMinor version;
     };
 
     src = fetchurl {
-      url = "mirror://gnome/sources/vala/${major}/${name}.tar.xz";
+      url = "mirror://gnome/sources/${pname}/${lib.versions.majorMinor version}/${pname}-${version}.tar.xz";
       inherit sha256;
     };
 
@@ -78,19 +76,26 @@
 
     nativeBuildInputs = [
       pkgconfig flex bison libxslt
-    ] ++ lib.optional (stdenv.isDarwin && (atLeast "0.38")) expat
+    ] ++ lib.optional (stdenv.isDarwin && (lib.versionAtLeast version "0.38")) expat
       ++ lib.optional disableGraphviz autoreconfHook # if we changed our ./configure script, need to reconfigure
       ++ extraNativeBuildInputs;
 
     buildInputs = [
       glib libiconv libintl
-    ] ++ lib.optional (atLeast "0.38" && withGraphviz) graphviz
+    ] ++ lib.optional (lib.versionAtLeast version "0.38" && withGraphviz) graphviz
       ++ extraBuildInputs;
 
     enableParallelBuilding = true;
 
     doCheck = false; # fails, requires dbus daemon
 
+    # passthru = {
+    #   updateScript = gnome3.updateScript {
+    #     packageName = pname;
+    #     attrPath = "${pname}_${lib.versions.major version}_${lib.versions.minor version}";
+    #   };
+    # };
+
     meta = with stdenv.lib; {
       description = "Compiler for GObject type system";
       homepage = https://wiki.gnome.org/Projects/Vala;
@@ -102,33 +107,28 @@
 
 in rec {
   vala_0_36 = generic {
-    major   = "0.36";
-    minor   = "18";
+    version = "0.36.18";
     sha256  = "0csb9skgy663y05kl813dsarzjbfcdsmx5fvz13p8gas3hycciq9";
   };
 
   vala_0_38 = generic {
-    major   = "0.38";
-    minor   = "10";
+    version = "0.38.10";
     sha256  = "1rdwwqs973qv225v8b5izcgwvqn56jxgr4pa3wxxbliar3aww5sw";
     extraNativeBuildInputs = [ autoconf ] ++ lib.optional stdenv.isDarwin libtool;
   };
 
   vala_0_40 = generic {
-    major   = "0.40";
-    minor   = "14";
+    version = "0.40.14";
     sha256  = "0llid9b9cgjcrcclc0pw2skkssb7br7b2clq9cql3p14dl94gki0";
   };
 
   vala_0_42 = generic {
-    major   = "0.42";
-    minor   = "6";
+    version = "0.42.6";
     sha256  = "14024gvs23q323fmd62hqd8jiypaxbjjvamyd782ixbhxmpz8x1p";
   };
 
   vala_0_44 = generic {
-    major   = "0.44";
-    minor   = "1";
+    version = "0.44.1";
     sha256 = "1qiglkgymws6a3m8xz1v8b3na165ywx2dbipp3bdg134bi0w3a4n";
   };
 
```

But even then there are some more problems, mainly the fact that the `updateScript`s are run in parallel. I suggest you use the patch above but not include the `updateScript` itself until #59372 is merged.

Reply:
Yeah haha, I had my suspicions that it wouldn't work. Thanks for the patch, I just had to fix some small things!